instrumentation/camel-2.20/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsSpanAssertions.java

```diff
@@ -0,0 +1,187 @@
+      SpanDataAssert span,
+      int index,
+      String spanName,
+      String bucketName,
+      String method,
+      SpanData parentSpan) {
+    if (index == 0) {
+      span.hasNoParent();
+    } else {
+      span.hasParent(parentSpan);
+    }
+    span.hasName(spanName)
+        .hasAttributesSatisfyingExactly(
+            equalTo(stringKey("aws.agent"), "java-aws-sdk"),
+            satisfies(stringKey("aws.endpoint"), val -> val.isInstanceOf(String.class)),
+            equalTo(stringKey("rpc.system"), "aws-api"),
+            equalTo(stringKey("rpc.method"), spanName.substring(3)),
+            equalTo(stringKey("rpc.service"), "Amazon S3"),
+            equalTo(stringKey("aws.bucket.name"), bucketName),
+            equalTo(stringKey("http.method"), method),
+            equalTo(longKey("http.status_code"), 200),
+            satisfies(stringKey("http.url"), val -> val.isInstanceOf(String.class)),
+            equalTo(stringKey("net.protocol.name"), "http"),
+            equalTo(stringKey("net.protocol.version"), "1.1"),
+            satisfies(stringKey("net.peer.name"), val -> val.isInstanceOf(String.class)),
+            satisfies(
+                SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                val ->
+                    val.satisfiesAnyOf(
+                        v -> assertThat(v).isNull(), v -> assertThat(v).isInstanceOf(Long.class))),
+            satisfies(
+                stringKey("net.peer.port"),
+                val ->
+                    val.satisfiesAnyOf(
+                        v -> val.isInstanceOf(Number.class), v -> assertThat(v).isNull())));
+  }
+
+  static void sns(SpanDataAssert span, int index, String spanName, SpanData parentSpan) {
+    if (index == 0) {
+      span.hasNoParent();
+    } else {
+      span.hasParent(parentSpan);
+    }
+    span.hasName(spanName)
+        .hasKind(CLIENT)
+        .hasAttributesSatisfyingExactly(
+            equalTo(stringKey("aws.agent"), "java-aws-sdk"),
+            satisfies(stringKey("aws.endpoint"), val -> val.isInstanceOf(String.class)),
+            equalTo(stringKey("rpc.system"), "aws-api"),
+            equalTo(stringKey("rpc.method"), spanName.substring(4)),
+            equalTo(stringKey("rpc.service"), "AmazonSNS"),
+            equalTo(stringKey("http.method"), "POST"),
+            equalTo(longKey("http.status_code"), 200),
+            satisfies(stringKey("http.url"), val -> val.isInstanceOf(String.class)),
+            equalTo(stringKey("net.protocol.name"), "http"),
+            equalTo(stringKey("net.protocol.version"), "1.1"),
+            satisfies(stringKey("net.peer.name"), val -> val.isInstanceOf(String.class)),
+            satisfies(
+                stringKey("net.peer.port"),
+                val ->
+                    val.satisfiesAnyOf(
+                        v -> val.isInstanceOf(Number.class), v -> assertThat(v).isNull())),
+            satisfies(
+                SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
```

Comment:
This attribute did not appear in the old groovy assertions; did we start reporting it at some point?